java/daikon/chicory/Instrument24.java

```diff
@@ -0,0 +1,1720 @@
+
+  /**
+   * Instrument the current class.
+   *
+   * @param classBuilder for the class
+   * @param classModel for the class
+   * @param loader the class loader for the class
+   */
+  private void instrumentClass(
+      ClassBuilder classBuilder, ClassModel classModel, ClassLoader loader) {
+
+    // Save constant pool builder for later use.
+    poolBuilder = classBuilder.constantPool();
+    classInfo = new ClassInfo(binaryClassName, loader);
+
+    debugInstrument.log("Class Attributes:%n");
+    for (java.lang.classfile.Attribute<?> a : classModel.attributes()) {
+      debugInstrument.log("  %s%n", a);
+    }
+
+    // Have each non-void method save its result in a local variable before returning.
+    instrument_all_methods(classModel, classBuilder);
+
+    // Remember any constant static fields.
+    List<FieldModel> fields = classModel.fields();
+    for (FieldModel fm : fields) {
+      Optional<ConstantValueAttribute> cva = fm.findAttribute(Attributes.constantValue());
+      if (cva.isPresent()) {
+        String name = fm.fieldName().stringValue();
+        String value = formatConstantDesc(cva.get().constant().constantValue());
+        debugInstrument.log("  Constant field: %s, value: %s%n", name, value);
+        classInfo.staticMap.put(name, value);
+      }
+    }
+
+    if (Chicory.checkStaticInit && !hasClinit) {
+      // If no clinit method, we need to add our own.
+      classBuilder.withMethod(
+          "<clinit>",
+          MethodTypeDesc.of(CD_void),
+          ClassFile.ACC_STATIC,
+          methodBuilder ->
+              methodBuilder.withCode(
+                  codeBuilder -> {
+                    codeBuilder
+                        .ldc(binaryClassName)
+                        .invokestatic(
+                            runtimeCD, "initNotify", MethodTypeDesc.of(CD_void, CD_String))
+                        .return_();
+                  }));
+    }
+  }
+
+  /**
+   * Adds a call (or calls) to the Chicory Runtime {@code initNotify} method into a given method.
+   * Clients pass the class static initializer {@code <clinit>} as the method.
+   *
+   * @param mgen the method to modify, typically the class static initializer
+   */
+  private void addInvokeToClinit(MethodGen24 mgen) {
+
+    try {
+      List<CodeElement> il = mgen.getInstructionList();
+      ListIterator<CodeElement> li = il.listIterator();
+      while (li.hasNext()) {
+
+        CodeElement inst = li.next();
+
+        // back up iterator to point to 'inst'
+        li.previous();
+
+        // Get the translation for this instruction (if any)
+        List<CodeElement> new_il = xform_clinit(inst);
+
+        // insert code prior to 'inst'
+        for (CodeElement ce : new_il) {
+          li.add(ce);
+        }
+
+        // skip over 'inst' we just inserted new_il in front of
+        li.next();
+      }
+    } catch (Exception e) {
+      System.err.printf("Unexpected exception encountered: %s", e);
+      e.printStackTrace();
```

Comment:
Again, the first line prints to stderr and the second to stdout.  Is that intentional?  If so, what is the purpose?

Reply:
as noted before - default is stderr